Add determine version documentation

another-rex opened the pull request:
Add the API documentation for DetermineVersion experimental endpoint.

hayleycd commented on docs/post-v1experimental-determineversion.md:
```diff
@@ -0,0 +1,95 @@
+title: POST /v1experimental/determineversion
+permalink: /post-v1experimental-determineversion/
+parent: API
+nav_order: 2
```
```suggestion
nav_order: 5
```

hayleycd commented on docs/post-v1experimental-determineversion.md:
```diff
@@ -0,0 +1,95 @@
+---
+layout: page
+title: POST /v1experimental/determineversion
+permalink: /post-v1experimental-determineversion/
```
```suggestion
permalink: /post-v1-determineversion/
```
What do you think of making this the URL, so it does not need to change when it goes out of the experimental stage?

another-rex replied:
Good point!

hayleycd commented on docs/post-v1experimental-determineversion.md:
```diff
@@ -0,0 +1,95 @@
+- TOC
+{:toc}
+</details>
+
```
```suggestion

## Experimental endpoint

This is a new, experimental API endpoint.  The purpose of the endpoint is to help determine the version of a given C/C++ library. It is difficult to know the correct version of C/C++ projects because there is not a centralized package manager within the ecosystem. This API endpoint helps bridge that gap. Once you have the likely version, you can use [POST v1/query](post-v1-query.md) or [POST v1/queryset](post-v1-queryset.md) to search for vulnerabilities. 

As this is an experimental feature, we would love to hear about your experience using it. If you give this a try, please consider [opening an issue](https://github.com/google/osv.dev/issues/new) and letting us know about any pain points or highlights. 
```

@another-rex I think we need something like this paragraph on this page. We want to be clear that it is experimental, explain a little about why it is needed, and solicit feedback. I am open to changing up the wording, but I think it is important to provide some context.

another-rex replied:
Thanks! Agreed, this is important context.

hayleycd commented on docs/post-v1experimental-determineversion.md:
```diff
@@ -0,0 +1,95 @@
+---
+layout: page
+title: POST /v1experimental/determineversion
```
[Here](https://raw.githubusercontent.com/hayleycd/image-library/main/osv-docs/navbar.png) is what it currently looks like.

hayleycd commented:
Generally this also needs information on how to get the hashes, how many files are necessary to include, etc.

another-rex commented:
Moved some wording around, added specific instructions for how to build the hashes to be sent to the endpoint, and linked the example tooling in the repository.

hayleycd reviewed (changes requested):
This is looking stronger, but I have remaining questions about the indexer-api-caller tool and how it would be used by a typical user. Let's chat to be sure we are on the same page.

oliverchang commented on docs/post-v1-determineversion.md:
```diff
@@ -0,0 +1,120 @@
+
+Given the source code of C/C++ libraries, this endpoint attempts to find the closest library and version.
+
+The purpose of the endpoint is to help determine the version of a given C/C++ library. It is difficult to know the correct version of C/C++ projects because there is not a centralized package manager within the ecosystem. This API endpoint helps bridge that gap. Once you have the likely version, you can use [POST v1/query](post-v1-query.md) or [POST v1/queryset](post-v1-queryset.md) to search for ...
```
Can we provide some info on which repos we index? And our future plans to extend coverage?

hayleycd replied:
Added. Thanks!

hayleycd commented:
[Docs preview](https://another-rex.github.io/osv.dev/post-v1-determineversion/)

hayleycd commented:
This is ready to go content-wise, but something broke in the github pages settings. I converted to draft until I can figure out what happened.

hayleycd commented:
Please merge https://github.com/google/osv.dev/pull/1396 first to avoid breaking

hayleycd commented:
@another-rex This is ready to go, but I took a look at the merge conflicts and determined that you would be the better person to resolve them.
